src/app/services/baw-api/account.service.spec.ts

```diff
@@ -8,214 +8,378 @@ import { Subject } from "rxjs";
+
+    it("should handle response with random user model id", fakeAsync(() => {
+      const model = new User({
+        id: 5,
+        userName: "username"
+      });
+
+      createSuccess("/user_accounts/5", model);
+
+      service.show(model).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
+
+    it("should handle response with random user id", fakeAsync(() => {
+      const model = new User({
+        id: 5,
+        userName: "username"
+      });
+
+      createSuccess("/user_accounts/5", model);
+
+      service.show(5).subscribe((_model: User) => {
+        expect(_model).toBeTruthy();
+        expect(_model).toEqual(model);
+      }, shouldNotFail);
+
+      tick(100);
+    }));
+
+    it("should handle error", fakeAsync(() => {
+      createError("apiShow", "/user_accounts/1", errorResponse);
+
+      service.show(1).subscribe(shouldNotSucceed, (err: ApiErrorDetails) => {
         expect(err).toBeTruthy();
-        expect(err).toEqual({
-          status: 401,
-          message: "Unauthorized"
-        } as ApiErrorDetails);
-      }
-    );
-
-    tick(100);
-  }));
-
-  it("should handle error with info", fakeAsync(() => {
-    spyOn<any>(service, "show").and.callFake(
-      (path: string, filters: Filters) => {
-        expect(path).toBe("/user_accounts/1");
-        expect(filters).toEqual(null);
-        const subject = new Subject<User>();
+        expect(err).toEqual(errorResponse);
+      });
 
-        setTimeout(() => {
-          subject.error({
-            status: 422,
-            message: "Record could not be saved",
-            info: {
-              name: ["has already been taken"],
-              image: [],
-              image_file_name: [],
-              image_file_size: [],
-              image_content_type: [],
-              image_updated_at: []
-            }
-          } as ApiErrorDetails);
-        }, 50);
+      tick(100);
+    }));
 
-        return subject;
-      }
-    );
-
-    service.show(1).subscribe(
-      () => {
-        expect(true).toBeFalsy("Service should not return data");
-      },
-      (err: ApiErrorDetails) => {
+    it("should handle error with info", fakeAsync(() => {
+      createError("apiShow", "/user_accounts/1", errorInfoResponse);
+
+      service.show(1).subscribe(shouldNotSucceed, (err: ApiErrorDetails) => {
         expect(err).toBeTruthy();
-        expect(err).toEqual({
-          status: 422,
-          message: "Record could not be saved",
-          info: {
-            name: ["has already been taken"],
-            image: [],
-            image_file_name: [],
-            image_file_size: [],
-            image_content_type: [],
-            image_updated_at: []
-          }
-        } as ApiErrorDetails);
-      }
-    );
-
-    tick(100);
-  }));
+        expect(err).toEqual(errorInfoResponse);
+      });
+
+      tick(100);
+    }));
+  });
```

Comment:
Generally, this test file is a lot simpler. Well done.

I'm not sure we need to test every combination you're testing here, especially for every service. But for now, good as is.